internal/shim-sgx/src/handler/enarx.rs

```diff
@@ -1,9 +1,35 @@
 // SPDX-License-Identifier: Apache-2.0
 
+use crate::sgx_uarch::REPORT;
+use crate::sgx_uarch::TARGETINFO;
+use core::arch::asm;
 use sallyport::syscall::{BaseSyscallHandler, EnarxSyscallHandler, SGX_QUOTE_SIZE, SGX_TECH};
 use sallyport::untrusted::{UntrustedRef, UntrustedRefMut, ValidateSlice};
 use sallyport::Reply;
 
+/// Invoke EREPORT
+#[allow(dead_code)]
+fn enclu_ereport(targetinfo: &TARGETINFO, reportdata: &[u8; 64]) -> crate::sgx_uarch::REPORT {
+    const EREPORT: usize = 0;
+
+    let mut report = core::mem::MaybeUninit::<REPORT>::uninit();
+
+    unsafe {
+        asm!(
+            "xchg       {RBX}, rbx",
+            "enclu",
+            "mov        rbx, {RBX}",
+
+            RBX = inout(reg) targetinfo => _,
+            in("rax") EREPORT,
+            in("rcx") reportdata.as_ptr(),
+            in("rdx") report.as_mut_ptr(),
```

Comment:
Either clean up this function now or file a debt issue.

Reply:
I create a deb issue, and my reasoning is:

- Before end-to-end my past experience test that it does not hurt to go over the top with not changing semantics (even obvious changes).
- It's a good first issue, right?